### PR TITLE
WIP: supports Ether and any ERC20 token

### PR DESCRIPTION
**Pull request by islishude**

### CODE REVIEW

**Review by AaronZgl — APPROVED** [review]
done

**Comment by xdefilab** [comment]
done

File: contracts/XHalfLife.sol [hunk 1]
```diff
@@ -106,30 +123,95 @@ contract XHalfLife is ReentrancyGuard {
...
+        createStreamPreflight(recipient, depositAmount, startBlock, kBlock)
+        returns (uint256)
+    {
+        require(token.isContract(), "not contract");
+        token.safeTransferFrom(msg.sender, address(this), depositAmount);
+
+        uint256 streamId = nextStreamId;
+        {
+            uint256 denom = 10**uint256(IERC20(token).decimals());
+            require(denom >= 10**6, "token decimal too low");
```
Comment (AaronZgl) [inline contracts/XHalfLife.sol:143]: Is this judgment necessary？

Comment (islishude) [reply contracts/XHalfLife.sol:143]: yes,effectsValue is 0.0004

Comment (islishude) [reply contracts/XHalfLife.sol:143]: effective withdrawable value: 0.0001

**Pull request closed without merging**